cheatsheets/Authentication_Cheat_Sheet.md

```diff
@@ -171,11 +171,58 @@ The application may return a different [HTTP Error code](https://developer.mozil
+
+- A new browser/device or IP address.
+- An unusual country or location.
+- Specific countries that are considered untrusted.
+- An IP address that appears on known blacklists.
+- An IP address that has tried to login to multiple accounts.
+- A login attempt that appears to be scripted rather than manual.
+
+Additionally, for enterprise applications, known trusted IP ranges could be added to a whitelist so that MFA is not required when users connect from these ranges.
+
+### Account Lockout
+
+The most common protection against these accounts is to implement account lockout, which prevents any more login attempts for a period after a certain number of failed logins.
+
+The counter of failed logins should be associated with the account itself, rather then the source IP address, in order to prevent an attacker making login attempts from a large number of different IP addresses. There are a number of different factors that should be considered when implementing an account lockout policy in order to find a balance between security and usability:
+
+- The number of failed attempts before the account is locked out (lockout threshold).
+- The time period that these attempts must occur within (observation window).
+- How long the account is locked out for (lockout duration).
+
+Rather than implementing a fixed lockout duration (e.g, ten minutes), some applications use and exponential lockout, where the lockout duration starts as a very short period (e.g, one second), but doubles after each failed login attempt.
+
+When designing an account lockout system, care must be taken to prevent it being used to cause a denial of service by locking out other users' accounts. One way this could be performed is to allow the user of the forgotten password functionality to login, even if the account is locked out.
+
+### CAPTCHA
+
+The use of an effective CAPTCHA can help to prevent automated login attempts against accounts. However, many CAPTCHA implementations have weaknesses that allow them to be solved using automated techniques, or can be outsourced to services which can solve them. As such, the use of CAPTCHA should be viewed as a defence-in-depth control to make brute-force attacks more time consuming and expensive, rather than as a preventative.
+
+It may be more user-friendly to only require a CAPTCHA be solved after a small number of failed login attempts, rather than requiring it from the very first login.
+
+### Security Questions and Memorable Words
+
+The additional of a security question or memorable word can also help protect against automated attacks, especially when the user is asked to enter a number of randomly chosen characters from the word. It should be noted that this does **not** constitute multi-factor authentication, as both factors are the same (something you know).
```

Comment:
- The `addition`.
- What do you think about adding the risk of having weak security questions?

Reply:
Added a link to the security questions CS